cli/azd/pkg/azdo/service_connection.go

```diff
@@ -50,6 +52,19 @@ func authorizeServiceConnectionToAllPipelines(
 	return nil
 }
 
+func ServiceConnection(
+	ctx context.Context,
+	connection *azuredevops.Connection,
+	projectId string, serviceConnectionName *string) (*serviceendpoint.ServiceEndpoint, error) {
+
+	client, err := serviceendpoint.NewClient(ctx, connection)
+	if err != nil {
+		return nil, fmt.Errorf("creating new azdo client: %w", err)
+	}
+
+	return serviceConnectionExists(ctx, &client, &projectId, serviceConnectionName)
```

Comment:
NIT: Sort of weird that this `Exists` method returns the underlying resource instead of a `bool`, and also a little strange perhaps that it ends up creating a service connection if it doesn't exist.  Perhaps `getOrCreateServiceConnection` would be a better name long term?

Reply:
agree.  This is 100% legacy code from when ado was initially implemented.

I added this new exported function `ServiceConection()` while I was looking at the expected structure for a Service Connection in ADO (since docs are terrible!!).

I will just delete this method as we don't really need it